From 0599f38e6b19ee737ae0da90f82762210186a536 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <tim.stahlhut@gmail.com>
Date: Fri, 4 Sep 2015 15:23:28 -0400
Subject: [PATCH 2/3] * cbp_wx28_32: Added added lib folder to some CB
 projects. (Thanks stahta01)

Made sure that all the .a files are under lib folder.
Removed some unneeded search folders.
Removed some unneeded libs.
---
 src/CodeBlocks.cbp                                 | 61 +++++++++++-----------
 src/plugins/contrib/SpellChecker/SpellChecker.cbp  |  5 +-
 src/plugins/contrib/ToolsPlus/ToolsPlus.cbp        |  1 -
 src/plugins/contrib/cb_koders/cb_koders.cbp        |  2 +-
 src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp |  4 +-
 src/plugins/contrib/help_plugin/help-plugin.cbp    |  9 ++--
 src/plugins/contrib/source_exporter/Exporter.cbp   |  6 +--
 .../contrib/wxContribItems/wxContribItems.cbp      | 17 +++---
 src/plugins/contrib/wxSmith/wxSmith.cbp            |  2 +-
 src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp   |  3 +-
 src/tools/cb_share_config/cb_share_config.cbp      |  2 +-
 11 files changed, 56 insertions(+), 56 deletions(-)

diff --git a/src/CodeBlocks.cbp b/src/CodeBlocks.cbp
index 6b8b55a..d09a777 100644
--- a/src/CodeBlocks.cbp
+++ b/src/CodeBlocks.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="tinyXML">
-				<Option output="base/tinyxml/txml" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/txml" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -26,7 +26,7 @@
 				<Linker>
 					<Add option="-s" />
 					<Add library="txml" />
-					<Add directory="base/tinyxml" />
+					<Add directory="lib" />
 				</Linker>
 			</Target>
 			<Target title="ConsoleRunner">
@@ -40,7 +40,7 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel">
-				<Option output="sdk/scripting/lib/squirrel" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sdk/squirrel" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -51,9 +51,9 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel std lib">
-				<Option output="sdk/scripting/lib/sqstdlib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sdk/sqstdlib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;" />
+				<Option external_deps="lib/sdk/libsquirrel.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -63,9 +63,9 @@
 				</Compiler>
 			</Target>
 			<Target title="SqPlus">
-				<Option output="sdk/scripting/lib/sqplus" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sdk/sqplus" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;" />
+				<Option external_deps="lib/sdk/libsquirrel.a;lib/sdk/libsqstdlib.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -75,7 +75,7 @@
 				</Compiler>
 			</Target>
 			<Target title="scintilla">
-				<Option output="devel/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sdk/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -101,7 +101,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxpropgrid">
-				<Option output="devel/wxpropgrid" prefix_auto="1" extension_auto="1" />
+				<Option output="devel/wxpropgrid" imp_lib="lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -123,9 +123,9 @@
 				</Linker>
 			</Target>
 			<Target title="sdk">
-				<Option output="devel/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option output="devel/codeblocks" imp_lib="lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="base/tinyxml/libtxml.a;sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;sdk/scripting/lib/libsqplus.a;devel/libwxscintilla_cb.a;" />
+				<Option external_deps="lib/libtxml.a;lib/sdk/libsquirrel.a;lib/sdk/libsqstdlib.a;lib/sdk/libsqplus.a;lib/sdk/libwxscintilla_cb.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --verbose -p debug" />
@@ -163,9 +163,9 @@
 					<Add library="sqstdlib" />
 					<Add library="squirrel" />
 					<Add library="gdi32" />
+					<Add directory="lib" />
+					<Add directory="lib/sdk" />
 					<Add directory="devel" />
-					<Add directory="sdk/scripting/lib" />
-					<Add directory="sdk/propgrid" />
 				</Linker>
 				<ExtraCommands>
 					<Add before="build_tools\autorevision\autorevision +wx +int +t . include/autorevision.h" />
@@ -178,7 +178,7 @@
 			<Target title="src">
 				<Option output="devel/codeblocks" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="0" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -234,7 +234,7 @@
 			<Target title="Abbreviations">
 				<Option output="devel/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -267,7 +267,7 @@
 			<Target title="AStyle">
 				<Option output="devel/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -299,7 +299,7 @@
 			<Target title="Autosave">
 				<Option output="devel/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -327,7 +327,7 @@
 				</ExtraCommands>
 			</Target>
 			<Target title="Compiler depslib">
-				<Option output="plugins/compilergcc/depslib/depslib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/plugins/compilergcc/depslib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -342,7 +342,7 @@
 			<Target title="Compiler">
 				<Option output="devel/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;plugins/compilergcc/depslib/libdepslib.a;" />
+				<Option external_deps="lib/libcodeblocks.a;lib/plugins/compilergcc/libdepslib.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -367,7 +367,7 @@
 					<Add library="depslib" />
 					<Add library="wxpropgrid" />
 					<Add directory="devel" />
-					<Add directory="plugins/compilergcc/depslib" />
+					<Add directory="lib/plugins/compilergcc" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="cmd /c if not exist devel\share\CodeBlocks mkdir devel\share\CodeBlocks" />
@@ -381,7 +381,7 @@
 			<Target title="Debugger">
 				<Option output="devel/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -413,7 +413,7 @@
 			<Target title="Code-completion">
 				<Option output="devel/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -449,7 +449,7 @@
 			<Target title="Class wizard">
 				<Option output="devel/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -479,7 +479,7 @@
 			<Target title="Default MIME handler">
 				<Option output="devel/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -509,7 +509,7 @@
 			<Target title="Occurrences Highlighting">
 				<Option output="devel/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -539,7 +539,7 @@
 			<Target title="Open files list">
 				<Option output="devel/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -569,7 +569,7 @@
 			<Target title="Projects-workspaces importer">
 				<Option output="devel/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -605,7 +605,7 @@
 			<Target title="Scripted wizard">
 				<Option output="devel/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -637,7 +637,7 @@
 			<Target title="To-do">
 				<Option output="devel/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -673,7 +673,7 @@
 			<Target title="XP look &amp; feel">
 				<Option output="devel/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -738,7 +738,6 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="base/tinyxml" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="base/tinyxml/tinystr.cpp">
diff --git a/src/plugins/contrib/SpellChecker/SpellChecker.cbp b/src/plugins/contrib/SpellChecker/SpellChecker.cbp
index 241ae47..dfdc534 100644
--- a/src/plugins/contrib/SpellChecker/SpellChecker.cbp
+++ b/src/plugins/contrib/SpellChecker/SpellChecker.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="hunspell">
-				<Option output="../../../devel/hunspell" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/plugins/contrib/SpellChecker/hunspell" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/SpellChecker/hunspell" />
 				<Option type="2" />
@@ -34,7 +34,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxSpellChecker">
-				<Option output="../../../devel/wxSpellChecker" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/plugins/contrib/SpellChecker/wxSpellChecker" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/SpellChecker/wxSpellChecker" />
 				<Option type="2" />
@@ -91,6 +91,7 @@
 					<Add library="hunspell" />
 					<Add library="codeblocks" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add directory="../../../lib/plugins/contrib/SpellChecker" />
 					<Add directory="../../../devel" />
 					<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)" />
 				</Linker>
diff --git a/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp b/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
index 002d635..5692c34 100644
--- a/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
+++ b/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
@@ -18,7 +18,6 @@
 				</Compiler>
 				<Linker>
 					<Add library="codeblocks" />
-					<Add library="wxscintilla_cb" />
 					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
 					<Add directory="../../../devel" />
diff --git a/src/plugins/contrib/cb_koders/cb_koders.cbp b/src/plugins/contrib/cb_koders/cb_koders.cbp
index ef561b0..fdbc8b7 100644
--- a/src/plugins/contrib/cb_koders/cb_koders.cbp
+++ b/src/plugins/contrib/cb_koders/cb_koders.cbp
@@ -10,7 +10,7 @@
 			<Target title="default">
 				<Option output="../../../devel/share/CodeBlocks/plugins/cb_koders.dll" prefix_auto="0" extension_auto="0" />
 				<Option object_output="../../../.objs/plugins/contrib/cb_koders" />
-				<Option external_deps="../../../devel/libcodeblocks.a;../../../devel/libwxscintilla_cb.a;" />
+				<Option external_deps="../../../lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp b/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
index a691278..452e4a9 100644
--- a/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
+++ b/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="libbz2">
-				<Option output="bzip2/libbz2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/plugins/contrib/devpak_plugin/libbz2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/devpak_plugin" />
 				<Option type="2" />
@@ -48,6 +48,7 @@
 				<Linker>
 					<Add option="-mthreads" />
 					<Add library="bz2" />
+					<Add directory="../../../lib/plugins/contrib/devpak_plugin" />
 					<Add directory="../../../devel" />
 				</Linker>
 				<ExtraCommands>
@@ -75,7 +76,6 @@
 			<Add option="-Wl,--enable-auto-import" />
 			<Add library="codeblocks" />
 			<Add library="wxmsw28$(WX_SUFFIX)" />
-			<Add directory="bzip2" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="bzip2/blocksort.c">
diff --git a/src/plugins/contrib/help_plugin/help-plugin.cbp b/src/plugins/contrib/help_plugin/help-plugin.cbp
index 7015570..515fb0e 100644
--- a/src/plugins/contrib/help_plugin/help-plugin.cbp
+++ b/src/plugins/contrib/help_plugin/help-plugin.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="bzip2">
-				<Option output="bzip2/libbzip2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/plugins/contrib/help_plugin/libbzip2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/help_plugin/bzip2" />
 				<Option type="2" />
@@ -22,7 +22,7 @@
 				</Linker>
 			</Target>
 			<Target title="zlib">
-				<Option output="zlib/libz.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/plugins/contrib/help_plugin/libz.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/help_plugin/zlib" />
 				<Option type="2" />
@@ -64,8 +64,9 @@
 				</Compiler>
 				<Linker>
 					<Add option="-mthreads" />
-					<Add library="bzip2\libbzip2.a" />
-					<Add library="zlib\libz.a" />
+					<Add library="libbzip2.a" />
+					<Add library="libz.a" />
+					<Add directory="../../../lib/plugins/contrib/help_plugin" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel\share\CodeBlocks\help_plugin.zip manifest.xml *.xrc *.png" />
diff --git a/src/plugins/contrib/source_exporter/Exporter.cbp b/src/plugins/contrib/source_exporter/Exporter.cbp
index 7d2dfe9..0f8c8aa 100644
--- a/src/plugins/contrib/source_exporter/Exporter.cbp
+++ b/src/plugins/contrib/source_exporter/Exporter.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="wxPdfDocument">
-				<Option output="wxPdfDocument/lib/wxPdfDocument" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/plugins/contrib/source_exporter/wxPdfDocument" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/source_exporter" />
 				<Option type="2" />
@@ -58,7 +58,7 @@
 			<Target title="default">
 				<Option output="../../../devel/share/CodeBlocks/plugins/Exporter" prefix_auto="0" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/source_exporter" />
-				<Option external_deps="wxPdfDocument/lib/libwxPdfDocument.a;" />
+				<Option external_deps="../../../lib/plugins/contrib/source_exporter/libwxPdfDocument.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
@@ -78,7 +78,7 @@
 					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
 					<Add directory="../../../devel" />
-					<Add directory="wxPdfDocument/lib" />
+					<Add directory="../../../lib/plugins/contrib/source_exporter" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel\share\CodeBlocks\Exporter.zip manifest.xml" />
diff --git a/src/plugins/contrib/wxContribItems/wxContribItems.cbp b/src/plugins/contrib/wxContribItems/wxContribItems.cbp
index 8d732db..0991763 100644
--- a/src/plugins/contrib/wxContribItems/wxContribItems.cbp
+++ b/src/plugins/contrib/wxContribItems/wxContribItems.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxFlatNoteBook">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxflatnotebook" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxflatnotebook" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxFlatNotebook" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -21,7 +21,7 @@
 			</Target>
 			<Target title="wxChartCtrl">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxchartctrl" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxchartctrl" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxChartCtrl" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -33,7 +33,7 @@
 			</Target>
 			<Target title="wxCustomButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxcustombutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxcustombutton" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxthings" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -45,7 +45,7 @@
 			</Target>
 			<Target title="wxImagePanel">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wximagepanel" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wximagepanel" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxImagePanel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -59,7 +59,7 @@
 			</Target>
 			<Target title="wxSpeedButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxspeedbutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxspeedbutton" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxSpeedButton" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -70,7 +70,7 @@
 			</Target>
 			<Target title="wxKWIC">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxkwic" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxkwic" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/KWIC" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -84,7 +84,7 @@
 			</Target>
 			<Target title="wxTreeList">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxtreelist" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxtreelist" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxtreelist" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -95,7 +95,7 @@
 			</Target>
 			<Target title="wxLED">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxled" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxled" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxled" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -131,7 +131,6 @@
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#wx.include)" />
 			<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
-			<Add directory="wxFlatNotebook/include" />
 		</Compiler>
 		<Linker>
 			<Add option="-Wl,--enable-auto-import" />
diff --git a/src/plugins/contrib/wxSmith/wxSmith.cbp b/src/plugins/contrib/wxSmith/wxSmith.cbp
index f0e656c..905895e 100644
--- a/src/plugins/contrib/wxSmith/wxSmith.cbp
+++ b/src/plugins/contrib/wxSmith/wxSmith.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxSmithLib">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxsmithlib" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxsmithlib" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="../../../devel" />
 				<Option object_output="../../../.objs/plugins/contrib/wxSmithLib" />
 				<Option type="3" />
diff --git a/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp b/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
index 89be967..f0719a5 100644
--- a/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
+++ b/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="MathPlot">
-				<Option output="../../../devel/MathPlot" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/plugins/contrib/wxSmithPlot/MathPlot" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/wxSmithPlot/MathPlot" />
 				<Option type="2" />
@@ -42,6 +42,7 @@
 					<Add library="wxsmithlib" />
 					<Add library="codeblocks" />
 					<Add library="wxpropgrid" />
+					<Add directory="../../../lib/plugins/contrib/wxSmithPlot" />
 					<Add directory="../../../devel" />
 				</Linker>
 				<ExtraCommands>
diff --git a/src/tools/cb_share_config/cb_share_config.cbp b/src/tools/cb_share_config/cb_share_config.cbp
index 44881f1..5dada02 100644
--- a/src/tools/cb_share_config/cb_share_config.cbp
+++ b/src/tools/cb_share_config/cb_share_config.cbp
@@ -46,7 +46,7 @@
 			<Add library="txml" />
 			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
-			<Add directory="../../base/tinyxml" />
+			<Add directory="../../lib" />
 		</Linker>
 		<Unit filename="app.cpp" />
 		<Unit filename="app.h" />
-- 
1.9.5.msysgit.1

